activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -106,6 +106,11 @@ def dangerous_attribute_method?(name) # :nodoc:
         ::ActiveRecord::AttributeMethods.dangerous_attribute_methods.include?(name.to_s)
       end
 
+      # Same as dangerous_attribute_method? but includes primary key methods.
+      def framework_reserved_method?(method_name) # :nodoc:
```

Comment:
We can't use `dangerous_attribute_method?` as it excludes primary key methods from the list and we want them to be treated as "dangerous" for the purpose of aliasing.

https://github.com/rails/rails/blob/a5a19ff5a5278081e5c7b251d68cf39300555d7f/activerecord/lib/active_record/attribute_methods/primary_key.rb#L98